xds/src/main/java/io/grpc/xds/InternalSharedXdsClientPoolProvider.java

```diff
@@ -30,4 +31,8 @@ private InternalSharedXdsClientPoolProvider() {}
   public static void setDefaultProviderBootstrapOverride(Map<String, ?> bootstrap) {
     SharedXdsClientPoolProvider.getDefaultProvider().setBootstrapOverride(bootstrap);
   }
+
+  public static ObjectPool<XdsClient> getOrCreate() throws XdsInitializationException {
```

Comment:
This is for grpc internal use only. If you need the `SharedXdsClientPoolProvider` I would rather make that public experimental. @ejona86 please advice the right way.

Reply:
There's a discussion ongoing on what approach to take. Depending on the answer, we may not need to expose this.

Reply:
I really don't want to expose these APIs as experimental. Even XdsClient, really. I'd much rather they are considered internal. They aren't for general use.